### PR TITLE
Extract PowerBI workspace & app info

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To better organize Power BI dashboards, reports, and datasets based on the workspace & app they belong to.

### 🤓 What?

As titled. Also refactored the tests for improved consistency.

### 🧪 Tested?

Manually tested against a production Power BI deployment.

### CODE REVIEW

File: metaphor/power_bi/extractor.py [hunk 1]
```diff
@@ -483,6 +510,16 @@ def map_wi_dashboards_to_dashboard(
...
+            pbi_info = PowerBIInfo(
+                power_bi_dashboard_type=PowerBIDashboardType.DASHBOARD,
+                workspace=PbiWorkspace(id=workspace.id, name=workspace.name),
+            )
+
+            if wi_dashboard.appId is not None:
+                app = app_map.get(wi_dashboard.appId)
+                if app is not None:
+                    pbi_info.app = PbiApp(id=app.id, name=app.name)
+
```
Comment (alyiwang) [inline metaphor/power_bi/extractor.py:522]: This part is very similar to the above line456. Maybe move it to a common function?

**Review by alyiwang — COMMENTED** [review]
Otherwise, LGTM. Great work!
Need to update the version.

File: metaphor/power_bi/extractor.py [hunk 2]
```diff
@@ -325,19 +342,21 @@ async def extract(self, config: PowerBIRunConfig) -> Collection[ENTITY_TYPES]:
...
+                    f"Fetching metadata from Power BI workspace ID: {workspace.id}"
                 )
 
                 try:
-                    self.map_wi_datasets_to_virtual_views(workspace_id, info.datasets)
-                    self.map_wi_reports_to_dashboard(workspace_id, info.reports)
-                    self.map_wi_dashboards_to_dashboard(workspace_id, info.dashboards)
+                    self.map_wi_datasets_to_virtual_views(workspace)
+                    self.map_wi_reports_to_dashboard(workspace, app_map)
+                    self.map_wi_dashboards_to_dashboard(workspace, app_map)
```
Comment (elic-eon) [inline metaphor/power_bi/extractor.py:359]: Thanks for refactoring.

**Review by elic-eon — APPROVED** [review]
LGTM

File: metaphor/power_bi/extractor.py (hunk 1 above)
Comment (mars-lan) [reply metaphor/power_bi/extractor.py]: Good point. Refactgored.

**Review by alyiwang — APPROVED** [review]
LGTM

**Pull request merged**